bin/oneoff/data_fix/bad_school_info_in_users

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+UPDATING = (ARGV[0].eql? "true").freeze
+puts "Called with UPDATING=#{UPDATING}"
+
+start_time = Time.now
+
+school_info_ids = ActiveRecord::Base.connection.execute(%Q{
+      SELECT DISTINCT t.id FROM (
+        SELECT MIN(id) AS id FROM school_infos WHERE validation_type = 'none' AND country = 'US'
+        AND school_type in ('public','charter')
+        GROUP BY school_type
+      UNION ALL
+        SELECT MIN(id) AS id FROM school_infos WHERE validation_type = 'none' AND country = 'US'
+        AND school_type in ('public','charter') AND state IS NOT NULL
+        GROUP BY school_type, state
+      UNION ALL
+        SELECT MIN(id) AS id FROM school_infos WHERE validation_type = 'none' AND country = 'US'
+        AND school_type in ('public','charter') AND state IS NOT NULL AND school_district_id IS NOT NULL
+        GROUP BY school_type, state, school_district_id
+      ) t JOIN school_infos si ON si.id = t.id
+      WHERE (si.school_id IS NOT NULL OR si.school_name IS NOT NULL)
+    }
+  ).each do |row|
+  row[0]
```

Comment:
The indentation is messed up here. The contents of the block should be indented 2 spaces.

Also what is this trying to do? I don't think it's doing what you intend. 

Why iterate and execute `row[0]` on each row? What is that supposed to do? Not the resulting `school_info_ids` is not the mapping of this block. For that you'd need `map` instead of `each`. This just executes the block on each row and returns the original set (each full row).

Reply:
If you just want an array of the Ids (position 0), you should be able to do `.pluck(0)`

Reply:
So, like: `ActiveRecord::Base.connection.execute(...).map.pluck(0)` ?

Reply:
no map, just `ActiveRecord::Base.connection.execute(...).pluck(0)`

Alternatively, map:
```ruby
ActiveRecord::Base.connection.execute(...).map do |row|
  row[0]
end
```

Reply:
Ex:

```
irb(main):014:0> ActiveRecord::Base.connection.execute('SELECT * from school_infos LIMIT 10').pluck(0)
=> [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

irb(main):015:0> ActiveRecord::Base.connection.execute('SELECT * from school_infos LIMIT 10').map{|r| r[0]}
=> [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```